src/Microsoft.NetCore.Analyzers/Core/Security/UseSecureCookiesASPNetCore.cs

```diff
@@ -0,0 +1,235 @@
+    {
+        internal static DiagnosticDescriptor DefinitelyUseSecureCookiesASPNetCoreRule = SecurityHelpers.CreateDiagnosticDescriptor(
+            "CA5382",
+            typeof(SystemSecurityCryptographyResources),
+            nameof(SystemSecurityCryptographyResources.DefinitelyUseSecureCookiesASPNetCore),
+            nameof(SystemSecurityCryptographyResources.DefinitelyUseSecureCookiesASPNetCoreMessage),
+            false,
+            helpLinkUri: null,
+            descriptionResourceStringName: nameof(SystemSecurityCryptographyResources.DoNotDisableSchUseStrongCryptoDescription),
+            customTags: WellKnownDiagnosticTagsExtensions.DataflowAndTelemetry);
+        internal static DiagnosticDescriptor MaybeUseSecureCookiesASPNetCoreRule = SecurityHelpers.CreateDiagnosticDescriptor(
+            "CA5383",
+            typeof(SystemSecurityCryptographyResources),
+            nameof(SystemSecurityCryptographyResources.MaybeUseSecureCookiesASPNetCore),
+            nameof(SystemSecurityCryptographyResources.MaybeUseSecureCookiesASPNetCoreMessage),
+            false,
+            helpLinkUri: null,
+            descriptionResourceStringName: nameof(SystemSecurityCryptographyResources.DoNotDisableSchUseStrongCryptoDescription),
+            customTags: WellKnownDiagnosticTagsExtensions.DataflowAndTelemetry);
+
+        public override ImmutableArray<DiagnosticDescriptor> SupportedDiagnostics => ImmutableArray.Create(
+                                                                                        DefinitelyUseSecureCookiesASPNetCoreRule,
+                                                                                        MaybeUseSecureCookiesASPNetCoreRule);
+
+        private static readonly ConstructorMapper constructorMapper = new ConstructorMapper(
+                                                                        ImmutableArray.Create<PropertySetAbstractValueKind>(
+                                                                            PropertySetAbstractValueKind.Flagged));
+
+        private static readonly PropertyMapperCollection PropertyMappers = new PropertyMapperCollection(
+            new PropertyMapper(
+                "Secure",
+                (ValueContentAbstractValue valueContentAbstractValue) =>
+                {
+                    switch (valueContentAbstractValue.NonLiteralState)
+                    {
+                        case ValueContainsNonLiteralState.No:
+                            // We know all values, so we can say Flagged or Unflagged.
+                            return valueContentAbstractValue.LiteralValues.Contains(true)
+                                ? PropertySetAbstractValueKind.Unflagged
+                                : PropertySetAbstractValueKind.Flagged;
+                        case ValueContainsNonLiteralState.Maybe:
+                            // We don't know all values, so we can say Flagged, or who knows.
```

Comment:
>Flagged [](start = 71, length = 7)

Unflagged

Reply:
Actually, I'm not sure if this approach will handle LiteralValues containing both true and false.  It's probably better to look for bad values.

---
In reply to: [284503253](https://github.com/dotnet/roslyn-analyzers/pull/2453#discussion_r284503253) [](ancestors = 284503253)

Reply:
Maybe a test case like this:
```csharp
bool secure = true;
if (flag) secure = false;
cookieOptions.Secure = secure;
```